### ISSUE TITLE
[Bug]: Accessibility - In multi-select drop down, NVDA is not giving the current state information

### ISSUE THREAD

**Issue by binhphan009** [microsoft/fluentui#29090]
### Library

React Northstar / v0 (@fluentui/react-northstar)

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
    Memory: 2.77 GB / 15.73 GB
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (116.0.1938.69)
    Internet Explorer: 11.0.19041.1566
```

### Are you reporting Accessibility issue?

Yes

### Reproduction

https://codesandbox.io/s/bug-accessibility-nvda-is-not-giving-the-current-state-information-2ppn6f?file=/index.js

## Actual Behavior
Repro Steps:
1. Open NVDA
2. Focus on to Multi select
3. Screen reader read "Clickable select your heroes menu button subMenu"
4. Screen reader does not announce current state information (collapsed)
<img width="465" alt="image" src="https://github.com/microsoft/fluentui/assets/105278576/f654db44-a265-47f8-a060-066c04abc76b">

5. Press Enter to expand the dropdown
6. Screen reader does not announce current state information (expanded)

<img width="466" alt="image" src="https://github.com/microsoft/fluentui/assets/105278576/f6837e52-61b7-44d1-bd62-a434829c7579">

## Expected Behavior
1. Screen reader should announce current state information (collapsed/expanded)
E.g: Clickable select your heroes menu **collapsed** button subMenu
2. Screen reader should announced the state of the combobox and dropdown after getting expanded and collapsed.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
